libkbfs/test_stallers.go

```diff
@@ -38,6 +39,166 @@ const (
 	StallableMDPutUnmerged           StallableMDOp = "PutUnmerged"
 )
 
+type naïveStallContext struct {
+	onStalled <-chan struct{}
+	unstall   chan<- struct{}
+	undoStall func()
+}
+
+// NaïveStaller is used to stall certain ops in BlockOps or MDOps . Unlike
+// StallBlockOp and StallMDOp which provides a way to precisely control which
+// particular op is stalled by passing in ctx with corresponding stallKey,
+// NaïveStaller simply stalls all instances of specified op.
+type NaïveStaller struct {
+	config Config
+
+	mu             *sync.RWMutex
+	blockOpsStalls map[StallableBlockOp]*naïveStallContext
+	mdOpsStalls    map[StallableMDOp]*naïveStallContext
+}
+
+// NewNaïveStaller returns a new NaïveStaller
+func NewNaïveStaller(config Config) *NaïveStaller {
+	return &NaïveStaller{
+		config:         config,
+		mu:             new(sync.RWMutex),
+		blockOpsStalls: make(map[StallableBlockOp]*naïveStallContext),
```

Comment:
you can also probably store the contexts/infos as value types instead of pointers.